activerecord/lib/active_record/connection_adapters/abstract_mysql_adapter.rb

```diff
@@ -732,11 +732,50 @@ def raw_execute(sql, name, async: false, allow_retry: false, uses_transaction: t
           log(sql, name, async: async) do
             with_raw_connection(allow_retry: allow_retry, uses_transaction: uses_transaction) do |conn|
               sync_timezone_changes(conn)
-              conn.query(sql)
+              result = conn.query(sql)
+              handle_warnings(sql)
+              result
             end
           end
         end
 
+        def handle_warnings(sql)
+          return if ActiveRecord.db_warnings_action == :ignore || @raw_connection.warning_count == 0
+
+          result = @raw_connection.query("SHOW WARNINGS")
+          result.each do |level, code, message|
+            warning = SqlWarning.new(message, code, level, sql)
+            next if warning_ignored?(warning)
+
+            case ActiveRecord.db_warnings_action
```

Comment:
My instinct is that this should be configurable per-connection, perhaps falling back to a global setting as the default. I don't have a strong argument for why that would be necessary, though -- maybe different adapters, or perhaps if you're early on in a 5.7 -> 8.0 switch or something?

Reply:
That would be logical, the problem is that AFAIK per db config is limited to YAML or URL in ENV, so it doesn't allow for procs and such.

What may make sense would be to pass the pool config or something to the callback to it can behave differently if needed or even just include the DB name.